### ISSUE TITLE
OperatorSpacing: Too much whitespace between short ternary

### ISSUE THREAD

**Issue by dereuromark** [squizlabs/PHP_CodeSniffer#3416]
**Describe the bug**
Too much whitespace between short ternary operators

**Code sample**
```php
        $x = $param ?  : null;
        $x = $param ?  ? null;
```

**Expected behavior**

```php
        $x = $param ?: null;
        $x = $param ?? null;
```

There should be no whitespace between `?:` and new `??` operators.
The sniffer currently does not find or fix this.

Also newlines are IMO never needed between those two, as they should be together for readability.

**Comment by jrfnl** [comment]
@dereuromark For the first one (short ternary), IIRC PSR-12 does not specify whether or not spacing is allowed between the `?` and the `:`, which is why it is not included in the PSR12 ruleset.

The `Squiz.WhiteSpace.OperatorSpacing` sniff would flag it, but expects one space and also looks at far more than just short ternaries, so is probably not what you want.

As for the second one, that's not a code style issue, that's a parse error and while PHPCS tries to not error out on parse errors, _fixing_ parse errors is outside of the scope.

If you want to, PHPCS can lint your code for parse errors and flag those using the `Generic.PHP.Syntax` sniff, but I'd recommend using a dedicated tool, like [PHP Parallel Lint](https://github.com/php-parallel-lint/PHP-Parallel-Lint/), instead.

**Comment by dereuromark** [comment]
Why is ?? Parse error? It is a null coalesence ternary, no?

**Comment by dereuromark** [comment]
Short ternary should also be in php9 ;)
